RecoLocalTracker/SiPixelRecHits/src/VVIObjF.cc

```diff
@@ -132,6 +133,88 @@ VVIObjF::VVIObjF(float kappa, float beta2, int mode) : mode_(mode) {
 
 }  // VVIObjF
 
+// ***************************************************************************************************************************************
+//! Constructor
+//! Set Vavilov parameter kappa and calculate the distribution fcn
+//! \param kappa - (input) Vavilov kappa parameter [0.01 (Landau-like) < kappa < 10. (Gaussian-like)]
+// ***************************************************************************************************************************************
+
+VVIObjF::VVIObjF(float kappa) {
+  const float xp[9] = {9.29, 2.47, .89, .36, .15, .07, .03, .02, 0.0};
```

Comment:
Seems like there is a bit of copy-pasting from the full constructor. Would it be possible to avoid it (to simplify maintenance and avoid bugs down the line), by using e.g. [delegating construtors](https://stackoverflow.com/questions/308276/can-i-call-a-constructor-from-another-constructor-do-constructor-chaining-in-c)?

Reply:
Hi @jpata I was thinking about doing this, but doesnt that defy the purpose?  If we have to construct the full constructor with the fixed parameters, that's still going to take similar time, no?

Reply:
I meant rather the other way around, to call the limited constructor from the full version (if that makes sense).

Is the full constructor actually used anywhere in CMSSW? Maybe it ought to be removed altogether.

Reply:
> I meant rather the other way around, to call the limited constructor from the full version (if that makes sense).

not entirely my cup of tea, but I don't immediately see how that would work, the functional form gets simplified in the case of the constructor with `kappa` alone. 

> Is the full constructor actually used anywhere in CMSSW? Maybe it ought to be removed altogether.

The full constructor should not be used anywhere in `cmssw` after this gets merged, but if I am not mistaken these files are used also by the pixel standalone simulation which doesn't live in cmssw (and perhaps that does need the full constructor, @tvami might confirm or disprove).   

> Would it be possible to avoid it (to simplify maintenance and avoid bugs down the line), 

About possible maintenance issues, as far as I understand there hasn't been any maintenance needed in many years, excepted extremely trivial one (cf history of this file: https://github.com/cms-sw/cmssw/commits/master/RecoLocalTracker/SiPixelRecHits/interface/VVIObjF.h), nor I think one would expect it (it's a bunch of mathematical expressions).

Reply:
I confirm Marco's comments, there is value in keeping the full constructor. I also confirm that the real content of this package hasn't really changed since 1993... it was just moved from Fortran to cpp and then optimized for better performance.

Reply:
Maybe then add a code comment then to each constructor implementation as an explicit warning to change both, if any changes are required to one (after all, now we are changing it, and who knows what will happen later).

Reply:
Sure I can do that!

Reply:
Since it's really just adding a comment, I hope you dont mind if I do this after the tests finished?

Reply:
Hi @jpata I added the comment in https://github.com/cms-sw/cmssw/pull/37533/commits/c72dd4590f4e5e05424e42b2ed8664936e0a9549